### ISSUE TITLE
Incorrect documentation & bug - monitoring_monitored_project [BETA]

### ISSUE THREAD

**Issue by drandell** [hashicorp/terraform-provider-google#10691]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

Terraform 1.0.10
Google Beta Provider - 4.2.1

### Affected Resource(s)

* google_monitoring_monitored_project

### Terraform Configuration Files

```tf
data "google_project" "proc_project" {
  project_id = var.ingest_project_id
}

data "google_project" "ops" {
  project_id = var.monitoring_project_id
}

resource "google_monitoring_monitored_project" "monitored-project" {
  provider = google-beta
  metrics_scope = var.monitoring_project_id
  name = var.ingest_project_id
}
```

Not sure if the correct place to put, but the documentation for the monitoring_monitored project example says to use the `name` of the project rather than the id/number. 
However, even if i use the project_id or number it fails with the below error. 
From the data object we get `projects/[PROJECT_ID]`, my guess would be that the initial projects/ is not being handled correctly. 
Using just two variables to pass in project_ids works (in the above example).

### Debug Output

```tf
 {
│     "@type": "type.googleapis.com/google.rpc.ErrorInfo",
│     "domain": "googleapis.com",
│     "metadata": {
│       "consumer": "projects/projects",
│       "service": "monitoring.googleapis.com"
│     },
│     "reason": "CONSUMER_INVALID"
│   }
│ ]
```

### Expected Behavior

It should add the metrics scope to my monitoring project.

### Actual Behavior

It fails to create to add the metric scope.

**Comment by megan07** [comment]
Hi @drandell ! Sorry you're experiencing this issue. I'm curious if you tried this:

```
data "google_project" "proc_project" {
  project_id = var.ingest_project_id
}

data "google_project" "ops" {
  project_id = var.monitoring_project_id
}

resource "google_monitoring_monitored_project" "monitored-project" {
  provider = google-beta
  metrics_scope = data.google_project.ops.name
  name = data.google_project.proc_project.name
}
```

it seemed to work for me?
Thanks!

**Comment by drandell** [comment]
Hi @megan07 , created with tf using just the project ids, i've deleted that
```tf
resource "google_monitoring_monitored_project" "monitored-project" {
      - create_time   = "2021-12-06T09:29:35.899276544Z" -> null
      - id            = "locations/global/metricsScopes/bt-tch-cdh-c-dev-ops/projects/bt-tch-cdh-c-dev-proc" -> null
      - metrics_scope = "ops_project_id" -> null
      - name          = "proc_project_id" -> null
    }
```

Recreating now with your snippet, the plan shows:
```tf
+ resource "google_monitoring_monitored_project" "monitored-project" {
      + create_time   = (known after apply)
      + id            = (known after apply)
      + metrics_scope = "[OPS DISPLAY NAME]"
      + name          = "[PROC DISPLAY NAME]"
    }
```

I get:
```
module.cloud_monitoring.google_monitoring_monitored_project.monitored-project: Creating...
╷
│ Error: Error creating MonitoredProject: failed to create a diff: failed to retrieve MonitoredProject resource: googleapi: Error 400: Request contains an invalid argument.
│ 
│   with module.cloud_monitoring.google_monitoring_monitored_project.monitored-project,
│   on ../modules/monitoring/main.tf line 37, in resource "google_monitoring_monitored_project" "monitored-project":
│   37: resource "google_monitoring_monitored_project" "monitored-project" {
```

**Comment by megan07** [comment]
@drandell the project names containing hyphens shouldn't matter, that should be fine. The invalid argument error is from the google side, are you able to provide debug logs?

**Comment by drandell** [comment]
[plan-debug.txt](https://github.com/hashicorp/terraform-provider-google/files/7663171/plan-debug.txt)
[apply-debug.txt](https://github.com/hashicorp/terraform-provider-google/files/7663184/apply-debug.txt)

Attached are the plan and apply debug logs :) @megan07

**Comment by megan07** [comment]
Got it! Thank you for that! So I would guess it's the spaces in the name that aren't working for you, and that makes sense. Looking into this more, the `name` returned is the projects "display name", and, in my test, our name and id is the same, which is why it works for me :). I will update these docs. I have tried `project_id` and `number` and both have worked for me.

To clarify, even if you use `project_id` or `number` this also fails with the consumer invalid error?
```
data "google_project" "proc_project" {
  project_id = var.ingest_project_id
}

data "google_project" "ops" {
  project_id = var.monitoring_project_id
}

resource "google_monitoring_monitored_project" "monitored-project" {
  provider = google-beta
  metrics_scope = data.google_project.ops.[project_id|number]
  name = data.google_project.proc_project.[project_id|number]
}
```
If these also fail, could you please send me the debug logs for these as well? (Sorry about that!)

**Comment by drandell** [comment]
No, I think you are probably right on that, I was using `data.google_project.ops.id` and not `.project_id`. Will confirm shortly

**Comment by kmunongonganyu** [comment]
Hello,
I am facing this issue too. The error message says 

`**Error: project: required field is not set
  on main.tf line 78, in data "google_project" "metric_scoping_project":
  78: data "google_project" "metric_scoping_project" {

Error: project: required field is not set

  on main.tf line 82, in data "google_project" "monitored_project":
  82: data "google_project" "monitored_project" {

➜  18_dashboards git:(OPDE-3424-metricscope) ✗** `

this is my code here 
`**resource "google_monitoring_monitored_project" "monitored_project" {
  metrics_scope = data.google_project.metric_scoping_project.project_id
  name          = data.google_project.monitored_project.project_id
  provider      = google-beta
}

data "google_project" "metric_scoping_project" {
  project_id     = var.metric_scoping_project_id
}

data "google_project" "monitored_project" {
  project_id     = var.monitored_project_id
}**`

**Comment by paulinaMorenoA** [comment]
Hello, I was facing same issue due to the documentation error on : [https://registry.terraform.io/providers/hashicorp/google-beta/latest/docs/resources/monitoring_monitored_project](url) 

Under **Argument Reference**  you can still see the reference to **project name** and no to **project ID**

> [name] - (Required) Immutable. The resource name of the MonitoredProject

**Comment by rileykarson** [comment]
Note: https://registry.terraform.io/providers/hashicorp/google-beta/latest/docs/resources/monitoring_monitored_project is misconfigured and works because the project's `name` and `project_id` match.

`name          = google_project.basic.name` should be `name          = google_project.basic.project_id`

**Comment by fancybear-dev** [comment]
@rileykarson thanks for the suggestion, had weird permission issues that shouldn't exist - your suggestion fixed the issue... Documentation should be updated.

**Comment by roaks3** [comment]
It looks like the example was updated correctly, but the `name` field documentation at https://registry.terraform.io/providers/hashicorp/google-beta/latest/docs/resources/monitoring_monitored_project still does not mention the short form options. It should be explicit that a project_id or project_number is allowed.
